tools/processresults.py

```diff
@@ -0,0 +1,317 @@
+"""Functions for processing CUES data in AppEngine JSON format into CSV format
+
+Has one public function, ProcessResults, used like this from interpreter:
+    ProcessResults('2015-01-06cuesDogfoodData.json', '2015-01-06-')
+"""
+
+import csv
+from datetime import datetime
+import json
+import re
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%S'
+DOGFOOD_START_DATE = datetime(2014, 12, 01, 0, 0, 0, 0)
+CONDITIONS = [
+    'ssl-overridable-proceed', 'ssl-overridable-noproceed',
+    'ssl-nonoverridable', 'malware-proceed', 'malware-noproceed',
+    'phishing-proceed', 'phishing-noproceed', 'extension-proceed',
+    'extension-noproceed']
+ATTRIBUTE_QUESTION_PREFIX = 'To what degree do each of the following'
+
+def ProcessResults(json_in_file, csv_prefix):
+  """Take results from AppEngine JSON file, process, and write to CSV file.
+
+  Results from the input JSON file will be filtered into the 9 experimental
+  conditions and written to CSV output files named in the format
+  <csv_prefix><condition>.csv. CSV column headers will be the meta data fields
+  date_received, date_taken, participant_id, and survey_type, plus the
+  questions asked of the participant (with [URL] replacing actual URLs in the
+  question text). Responses where questions were replaced with PLACEHOLDER
+  will be ignored and not copied to the CSV file.
+
+  Args:
+    json_in_file: File with raw AppEngine results in JSON format
+    csv_prefix: Prefix to output CSV files. These files will be named
+        in the format <csv_prefix><condition>.csv. For example, if
+        csv_prefix is '2015-01-15-' then one output file would be
+        named 2015-01-15-ssl-overridable-proceed.csv
+
+  Returns:
+    None
+  """
+  parsed_events = _ParseSurveyResults(json_in_file)
+  parsed_events = _DiscardResultsBeforeDate(parsed_events, DOGFOOD_START_DATE)
+
+  for c in CONDITIONS:
+    try:
+      filtered_results = _FilterByCondition(c, parsed_events)
+      canonical_index = _CanonicalizeQuestions(filtered_results)
+      _WriteToCsv(filtered_results, canonical_index, csv_prefix + c + '.csv')
+    # Print UnexpectedFormatException and continue, since they are usually
+    # due to lack of data for a condition.
+    except UnexpectedFormatException as e:
+      print 'Exception in %s: %s' % (c, e.value)
+  
+def _ParseSurveyResults(in_file):
+  with open(in_file, 'r') as json_file:
+    s = json_file.read()
+  parsed = _ParseJsonObjects(s)
+  events = filter(lambda x: x['survey_type'] != 'setup.js', parsed)
+  return events
+  
+def _ParseJsonObjects(s):
+  decoder = json.JSONDecoder()
+  objs = []
+  end = 0
+  while end < len(s) and end != -1:
+    obj, end = decoder.raw_decode(s, idx=end)
+    objs.append(obj)
+    end = s.find('{', end)
```

Comment:
I think this is because of https://github.com/GoogleChrome/experience-sampling/issues/131, but isn't it better to solve that issue rather than add custom parsing code here?